packages/core/src/performers/auto-performer/AutoPerformer.ts

```diff
@@ -157,6 +191,18 @@ export class AutoPerformer {
           }).summary
         : undefined;
 
+      if (this.cacheMode !== "disabled" && cacheKey) {
```

Comment:
the check implementation detail (`this.cacheMode !== "disabled"`) should be encapsulated in the cache handler